core/src/main/scala/chisel3/CompileOptions.scala

```diff
@@ -5,45 +5,7 @@ package chisel3
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox.Context
 
-trait CompileOptions {
-
-  /** Should Record connections require a strict match of fields.
-    *
-    * If true and the same fields aren't present in both source and sink, a MissingFieldException,
-    * MissingLeftFieldException, or MissingRightFieldException will be thrown.
-    */
-  val connectFieldsMustMatch: Boolean
-
-  /** When creating an object that takes a type argument, the argument must be unbound (a pure type). */
-  val declaredTypeMustBeUnbound: Boolean
-
-  /** If a connection operator fails, don't try the connection with the operands (source and sink) reversed. */
-  val dontTryConnectionsSwapped: Boolean
-
-  /** If connection directionality is not explicit, do not use heuristics to attempt to determine it. */
-  val dontAssumeDirectionality: Boolean
-
-  /** Check that referenced Data have actually been declared. */
-  val checkSynthesizable: Boolean
-
-  /** Require explicit assignment of DontCare to generate "x is invalid" */
-  val explicitInvalidate: Boolean
-
-  /** Should the reset type of Module be a Bool or a Reset */
-  val inferModuleReset: Boolean
-
-  /** If marked true, then any Module which consumes `inferModuleReset=false` must also mix in [[RequireSyncReset]] */
-  def migrateInferModuleReset: Boolean = false
-
-  /** If marked true, then any <> operation using legacy connection semantics will error and suggest using [[Connectable]] ops */
-  def migrateBulkConnections: Boolean = false
-
-  /** If marked true, then any := operation on mixed alignment bundles will error and suggest using [[Connectable]] ops */
-  def migrateMonoConnections: Boolean = false
-
-  /** Should connects emit as firrtl <= instead of <- */
-  def emitStrictConnects: Boolean = true
-}
+trait CompileOptions {}
```

Comment:
It can, but wouldn't all references to `CompileOptions` littered everywhere in chisel3 would need to do `@nowarn`, to keep chisel warning free?

Reply:
> but wouldn't all references to CompileOptions littered everywhere in chisel3 would need to do @nowarn, to keep chisel warning free?

No, because you could add it to the `-Wconf` in scalacOptions, but it doesn't really matter since we're removing this in https://github.com/chipsalliance/chisel/pull/3056